inst/include/common/fims_vector.hpp

```diff
@@ -0,0 +1,437 @@
+#ifndef FIMS_VECTOR_HPP
+#define FIMS_VECTOR_HPP
+
+#include "../interface/interface.hpp"
+
+namespace  fims {
+
+
+/**
+ * Wrapper class for std::vector types. If this file is compiled with -DTMB_MODEL,
+ * conversion operators are defined for TMB vector types.
+ * 
+ * All std::vector functions are copied over from the std library. While some of these may
+ * not be called explicitly in FIMS, they may be required to run other std library functions.
+ * 
+ */
+template<typename Type>
+class Vector{
+    std::vector<Type> vec_m;
+    
+    /**
+     * @brief friend comparison operator.
```

Comment:
what is meant by "friend" in this brief? Can more description be added so it is easier for the non-C++ expert to understand?